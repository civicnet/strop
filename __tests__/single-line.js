'use strict'

const StrOP = require('..');

const str = new StrOP('Single-line');

it('is empty', () => {
    expect(`${str``}`).toEqual('');
});

it('does not trim', () => {
    expect(`${str` `}`).toEqual(' ');
});

it('keeps raw text', () => {
    expect(`${str`\t`}`).toEqual('\\t');
});

it('interpolates', () => {
    expect(`${str`${42}`}`).toEqual('42');
});

it('keeps padding', () => {
    expect(`${str` before ${'&'} after `}`).toEqual(' before & after ');
});

it('is unchanged', () => {
    expect(`${str`${undefined} ${null}`}`).toMatchSnapshot();
    expect(`${str`${false} ${true}`}`).toMatchSnapshot();
    expect(`${str`${NaN} ${0} ${42} ${Infinity}`}`).toMatchSnapshot();
    expect(`${str`${''} ${' '} ${'text'}`}`).toMatchSnapshot();
});
