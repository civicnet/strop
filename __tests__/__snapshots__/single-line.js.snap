// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is unchanged 1`] = `"undefined null"`;

exports[`is unchanged 2`] = `"false true"`;

exports[`is unchanged 3`] = `"NaN 0 42 Infinity"`;

exports[`is unchanged 4`] = `"   text"`;
